core/src/main/scala/org/apache/spark/util/SparkExitCode.scala

```diff
@@ -45,6 +45,10 @@ private[spark] object SparkExitCode {
       OutOfMemoryError. */
   val OOM = 52
 
+  /** The default uncaught exception handler was reached and the exception was thrown by
+     TaskReaper. */
+  val KILLED_BY_TASK_REAPER = 53
```

Comment:
I did a bit of digging to figure out whether exit code 53 might have had any special meaning in Spark or in other systems and found https://stackoverflow.com/questions/45428145/do-exit-codes-and-exit-statuses-mean-anything-in-spark, which reminded me of the existence of `ExecutorExitCode.scala` where it's already used for a BlockManager-related exit condition:

https://github.com/apache/spark/blob/0f21df0b29cc18f0e0c7b12543f3a037e4032e65/core/src/main/scala/org/apache/spark/executor/ExecutorExitCode.scala#L30-L34

Given that the task reaper kill is an executor-specific behavior, I think we should be adding a new exit code in `ExecutorExitCode.scala`. Let's maybe use the next free code, 57, for that?

Reply:
Thanks for catching this! Will move KILLED_BY_TASK_REAPER to ExecutorExitCode and change the value to 57.